src/piecetype.cpp

```diff
@@ -65,6 +65,21 @@ auto Piece::fromWind(Wind w) -> Piece {
   }
 }
 
+auto Piece::formPiece(Piece::Type suit, uint8_t number) -> Piece {
+  auto valueAppend = number;
+  if (number == 1 || number == 9) {
+    valueAppend |= TERMINAL_BIT;
+  } else if (number == 0) {
+    valueAppend = 5 | RED_FIVE;
+  }
```

Comment:
I think we should move this functionality to handformer.cpp as it is closer to the notation spec. I would prefer a boolean flag for red fives here I think.

Reply:
Fixed in https://github.com/realliance/libmahjong/commit/0ca2dfbb82dcacdfd84b0b5e54a619783c0dfc7d